Lib/test/cov.py

```diff
@@ -0,0 +1,22 @@
+"""A minimal hook for gathering line coverage of the standard library."""
+
+import sys
+
+mon = sys.monitoring
+mon.use_tool_id(mon.COVERAGE_ID, "regrtest coverage")
+
+FileName = str
+LineNo = int
+Location = tuple[FileName, LineNo]
+COVERAGE: set[Location] = set()
+
+# `types` and `typing` not imported to avoid invalid coverage
+def add_line(
+    code: "types.CodeType",
+    lineno: int,
+) -> "typing.Literal[sys.monitoring.DISABLE]":
```

Comment:
I suppose that filenames are not normalized for the same reason: be able to register this code as soon as possible, before the "first" Python module (others than builtins and sys) are imported.